swap_meet/vendor.py

```diff
@@ -1,2 +1,85 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if not item in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+    
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            item_for_friend = self.remove(my_item)
+            other_vendor.add(item_for_friend)
+            
+            item_for_me = other_vendor.remove(their_item)
+            self.add(item_for_me)
+            return True
+    
+        return False
+    
+    def swap_first_item(self, other_vendor):
+        if len(self.inventory) == 0 or len(other_vendor.inventory) == 0:
+            return False
+        
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+
+        return self.swap_items(other_vendor, my_first_item, their_first_item)
+
+
+    def get_by_category(self, category):
+        category_list = []  
+        
+        for item in self.inventory:
+            if category == item.category:
+                category_list.append(item)
+        
+        return category_list
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
category_list = [item for item in self.inventory if item.get_category() == category]
return category_list
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!

Reply:
Also, using `get_category()` here would mean you wouldn't have to assign `self.category` in the `Item` initializer.